### PR TITLE
Lk pd 2518 shard metrics

### PR DESCRIPTION
**Pull request by ekiernan**
Added python script for calculating library-level metrics from STARsolo

### CODE REVIEW

**Review by kevinpalis — APPROVED** [review]
LGTM

File: tools/scripts/combine_shard_metrics.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    print("Calcuating metrics")
+    merge_pv[f"Reads Mapped to {counting}: Unique {counting}*n_reads"] = merge_pv[f'Reads Mapped to {counting}: Unique {counting}']*merge_pv['Number of Reads']
+    Sum_reads_mapped_unique_gene=merge_pv[f"Reads Mapped to {counting}: Unique {counting}*n_reads"].sum()
+    Total_reads_mapped_unique_gene=Sum_reads_mapped_unique_gene/n_reads
+    merge_pv["Reads Mapped to Genome: Unique*n_reads"] = merge_pv["Reads Mapped to Genome: Unique"]*merge_pv['Number of Reads']
+    merge_pv["Reads Mapped to Genome: Unique+Multiple*n_reads"] = merge_pv["Reads Mapped to Genome: Unique+Multiple"]*merge_pv['Number of Reads']
+    merge_pv["Q30 Bases in RNA read*n_reads"] = merge_pv["Q30 Bases in RNA read"]*merge_pv['Number of Reads']
+    merge_pv["Q30 Bases in CB+UMI*n_reads"] = merge_pv["Q30 Bases in CB+UMI"]*merge_pv['Number of Reads']
+    merge_pv["Reads With Valid Barcodes*n_reads"] = merge_pv["Reads With Valid Barcodes"]*merge_pv['Number of Reads']
+    Sequencing_Saturations_Total = 1-(merge_pv["yesUMIs"].sum()/merge_pv["yessubWLmatch_UniqueFeature"].sum())
```
Comment (khajoue2) [inline tools/scripts/combine_shard_metrics.py]: Can we rename these to be lower case? Sequencing_Saturations_Total -> sequencing_saturations_total

File: tools/scripts/combine_shard_metrics.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    Reads_mapped_Genome_unique=merge_pv["Reads Mapped to Genome: Unique*n_reads"].sum()/n_reads
```
Comment (khajoue2) [inline tools/scripts/combine_shard_metrics.py]: same comment, Reads_mapped_Genome_unique -> reads_mapped_genome_unique

File: tools/scripts/combine_shard_metrics.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    reads_mapped_antisense_to_gene=cells["exonicAS"].sum()+cells["intronicAS"].sum()
+    reads_exonic=cells["exonic"].sum()
+    reads_mapped_confidently_to_genome=cells["genomeU"].sum()
+    reads_mapped_confidently_to_intronic_regions=cells["intronic"].sum()
+    reads_mapped_confidently_to_transcriptome=cells["featureU"].sum()
+    data = {"Number_of_reads": [n_reads], 
+        "Sequencing_saturation": [Sequencing_Saturations_Total], 
+        "Fraction_of_unique_reads_mapped_to_genome": [Reads_mapped_Genome_unique],
+        "Fraction_of_unique_and_multiple_reads_mapped_to_genome": [Reads_mapped_Genome_unique_multi],
+        "Fraction_of_reads_with_Q30_bases_in_rna": [Q30_RNA],
```
Comment (khajoue2) [inline tools/scripts/combine_shard_metrics.py]: Could we change Fraction_of_reads_with_Q30_bases_in_rna to fraction_of_reads_with_Q30_bases_in_rna

**Review by khajoue2 — CHANGES_REQUESTED** [review]
For naming variables, it is good to stay consistent with _ and capital letters. We could use sequencing_saturation or sequencingSaturation but mixing the two styles are not recommended.

File: tools/scripts/combine_shard_metrics.py (hunk 1 above)
Comment (ekiernan) [reply tools/scripts/combine_shard_metrics.py]: this is just an internal variable that isn't outputted by the script. Do you still want it changed? I updated all the outputted CSV metrics to be lower case as suggested below.

Comment (ekiernan) [reply tools/scripts/combine_shard_metrics.py]: Oh, I saw your comment above! ok, I renamed

File: tools/scripts/combine_shard_metrics.py (hunk 3 above)
Comment (ekiernan) [reply tools/scripts/combine_shard_metrics.py]: yes, made that change

File: tools/scripts/combine_shard_metrics.py (hunk 2 above)
Comment (ekiernan) [reply tools/scripts/combine_shard_metrics.py]: yes, made that change

**Review by khajoue2 — APPROVED** [review]

**Pull request merged**